### ISSUE TITLE
Advanced Monitoring (for Quicksaves)

### ISSUE THREAD

**Issue by beppe9000** [MASGAU/MASGAU#100]
Many games just overwrite the quick save slot with the next one. also for many games the quick save is byte-identical to a normal save. thus i propose to have a feature that, when implemented will allow to:

-select a file to monitor for changes (from user or database) and allow the user (or database) to specify some rules to actually copy the file as if it were a new save and (if is the case) update registry keys/folder names/file names/ini files/etc...

In this way players of titles like Gothic 3(this is my case, but there are many) would happily have an in-game backup of their saves(wich will be backupped by the standard monitor functionality). 

Ps. For now I'm going to code this as a batch file/something similar only for gothic 3

**Comment by sanmadjack** [comment]
While what you're describing isn't implemented as of yet, couldn't MASGAU's versioning feature provide the extra quicksave backups you seek?

**Comment by beppe9000** [comment]
can you elaborate on this?

**Comment by sanmadjack** [comment]
On the settings tab there are versioning settings. This causes MASGAU to switch to a new archive every so often (you define the time period). So if you set it to 10 minutes, then you will end up with archives representing your saves as of every 10 minutes.

**Comment by beppe9000** [comment]
is this incremental?

**Comment by sanmadjack** [comment]
My instinct is to say yes, but this is a hard feature to describe. Basically, you set it to 10 minutes, then every 10 minutes it will make a copy of the archive, preserving the saves as they were at that moment.

**Comment by beppe9000** [comment]
so if i  set it to 20 minutes and play 60 i would get 3 archives like this:
save 1,2,3,4,5
save 1,2,3,4,5,6,7,8,9
save 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15

right? if it is so then it is not incremental.

incremental would be ending up with a single archive(updated every 20 minutes with new files):

save 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15

**Comment by sanmadjack** [comment]
Depends, are save 1,2,3,4 each saved in separate files/slots, or do they represent the same quicksave?

**Comment by beppe9000** [comment]
read: BackupArchive( save 1+2+3+4+5 ) etc

**Comment by beppe9000** [comment]
however i think that a "quick-save" feature to separate normal saves from quick saves in backups would be great.

**Comment by sanmadjack** [comment]
I mean are they saved in separate files outside of MASGAU's archive. It's important to remember that MASGAU only deals with saves at the file level. If given these files:

save1.sav
save2.sav
save3.sav

And each is written ten minutes apart then the resulting 3 archives will contain:

Archive 1: save1.sav
Archive 2: save1.sav, save2.sav
Archive 3: save1.sav, save2.sav, save3.sav

If all the saves are contained in one files, say saves.sav, then the three archives will only ever contain the one saves.sav file, but it will be a different version in each file.

Upon further reflection, an incremental pattern would be more like:
Archive 1: save1.sav
Archive 2: save2.sav
Archive 3: save3.sav

Because each archive would have to only store what changed since the last archive, which definitely isn't how MASGAU does it.

As for separating out quicksaves into a different file, that should be do-able as-is. MASGAU has the ability to separate game settings into a different archive, that could be re-used for this. It would make the <files> entry for a game look something like this:

``` xml
   <files>
      <include filename="save*.sav" />
   </files>
   <files type="QuickSaves">
      <include filename="quicksave.sav" />
   </files>
   <files type="Settings">
      <include filename="settings.ini" />
   </files>
```

Documentation on the XML format can be found at http://gamesave.info/xml_format.php#subtitle5

Granted, this doesn't answer your initial idea of MASGAU being able to transpose regular and quick saves automatically. MASGAU's purpose is a save backup tool, and that kind of feature is a more of a save management thing. While I agree it would be a cool, due to my limited availability I have to draw the line as to what features I'm willing to work on, and I tend to use the ruler of whether or not the feature is out of scope for MASGAU's intended purpose. If someone wanted this feature badly enough to put the time in and implement it themselves, then great, that's why MASGAU is open source in the first place (I'd want to discuss the implementation plan of course, but that's just logistics). As it is, the best I could say is that this would be relatively low priority, especially compared to things like the Mono conversion.

**Comment by beppe9000** [comment]
Yes, it is low priority. However I am starting to learn programming so maybe some time in the future i could be engaged in this. ;)
